### PR TITLE
alcom: init at 1.8.1

### PR DESCRIPTION
**Pull request by Scrumplex**
This is the experimental GUI counterpart of [vrc-get](https://github.com/vrc-get/vrc-get) (called vrc-get-gui in the repo).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Scrumplex** [comment]
CC @bddvlpr you might be interested in this ^^

**Comment by Scrumplex** [comment]
`nixfmt-check` is failing because of deps.nix. This is currently by design and there is no consensus on formatting these: https://github.com/NixOS/nixpkgs/pull/325053

**Comment by bddvlpr** [comment]
Interesting, thanks for the PR!

Seems to work as expected with a few minor visual bugs but those might very well be some Wayland moments.
image

**Review by bddvlpr — APPROVED** [review]

File: pkgs/by-name/al/alcom/package.nix [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  google-fonts,
+  lib,
+  libsoup,
+  nodejs,
+  npmHooks,
+  openssl,
+  pkg-config,
+  rustPlatform,
+  stdenv,
+  webkitgtk,
```
Comment (FliegendeWurst) [inline pkgs/by-name/al/alcom/package.nix:18]: Needs to be a specific version

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/al/alcom/package.nix [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    cargo-tauri.hook
+    dotnetSdk
+    nodejs
+    npmHooks.npmConfigHook
+    pkg-config
+  ];
+
+  buildInputs =
+    [ openssl ]
+    ++ lib.optionals stdenv.isLinux [
```
[the remark this answers is not in the document]
Comment (Scrumplex) [reply pkgs/by-name/al/alcom/package.nix:71]: good catch! I copied this from the tauri example iirc so it should probably be updated

File: pkgs/by-name/al/alcom/package.nix [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    dotnet restore "vrc-get-litedb/dotnet/vrc-get-litedb.csproj" \
+      -p:ContinuousIntegrationBuild=true \
+      -p:Deterministic=true
+  '';
+
+  passthru = {
+    inherit (dotnetBuild) fetch-deps;
+  };
+
+  meta = with lib; {
```
Comment (FliegendeWurst) [inline pkgs/by-name/al/alcom/package.nix:113]: Please set platforms = linux or fix the Darwin build issue
```
Undefined symbols for architecture x86_64:
 "____chkstk_darwin", referenced from:
```

File: pkgs/by-name/al/alcom/package.nix [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  passthru = {
+    inherit (dotnetBuild) fetch-deps;
+  };
+
+  meta = {
+    description = "Experimental GUI application to manage VRChat Unity Projects";
+    homepage = "https://github.com/vrc-get/vrc-get";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ Scrumplex ];
+    # aarch64-linux: Error failed to build app: Target aarch64-unknown-linux-gnu does not exist. Please run `rustup target list` to see the available targets.
```
Comment (FliegendeWurst) [inline pkgs/by-name/al/alcom/package.nix:118]: Did you build this via Nix? It seems odd that the rustup target is not available.

Comment (Scrumplex) [reply pkgs/by-name/al/alcom/package.nix:118]: I did. I am not quite sure why this error pops up.

**Review by FliegendeWurst — APPROVED** [review]

**Comment by anatawa12** [comment]
Hi! author of ALCOM here!

I found that alcom is incorrectly registered with version names of vrc-get.
vrc-get is incremented independently with ALCOM releases.

Could you fix version name?
if possible, could you update them?

Actually latest ALCOM, v0.1.17, is far new than ALCOM at the time vrc-get 1.8.2 is released.

**Pull request merged**